docs/Contributing.md

```diff
@@ -0,0 +1,130 @@
+
+* **DON'T** use abbreviations unless they are already well-known terms known by
+  users (e.g., "app", "info"), or are already required for use by developers (e.g.,
+  "min", "max", "args").  Examples of bad use would be `num_widgets` instead of
+  `widget_count`, and `opt_widgets` instead of `option_widgets` or `optional_widgets`.
+
+* **DON'T** use hard-coded magic numbers for things that have to be consistent
+  between different files.  Instead use a `#define` or an enum or const value, as appropriate.
+
+* **DON'T** use the same C function name with two different prototypes across
+  the project where possible.
+
+* **DON'T** use commented-out code, or code in an `#if 0` or equivalent.  Make sure all code is actually
+  built.
+
+Header Files
+------------
+
+* **DO** make sure any header file can be included directly, without requiring other
+  headers to be included first.  That is, any dependencies should be included within
+  the header file itself.
+
+* **DO** include system headers (with `<>`) before local headers (with `""`), and list them
+  in alphabetical order where possible.  This helps ensure there are not duplicate includes,
+  and also helps ensure that headers are usable directly.
+
+* **DO** use `#pragma once` in all header files, rather than using ifdefs to test for duplicate inclusion.
+
+Style Guide
+-----------
+
+### Automated Formatting with `clang-format`
+
+For all C/C++ files (`*.c`, `*.cpp` and `*.h`), we use `clang-format` (specifically
+version 3.6) to apply our code formatting rules. After modifying C/C++ files and
+before merging, be sure to run:
+
+```sh
+$ ./scripts/format-code
+```
+
+This allows us to apply formatting choices such as the use of [Allman style](
+http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces and the 80
+character column width consistently.
+
+Please stage the formatting changes with your commit, instead of making an extra
+"Format Code" commit. Your editor can likely be set up to automatically run
+`clang-format` across the file or region you're editing. See:
+
+- [clang-format.el](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/clang-format.el) for Emacs
+- [vim-clang-format](https://github.com/rhysd/vim-clang-format) for Vim
+- [vscode-cpptools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
+  for Visual Studio Code
+
+The [.clang-format](../.clang-format) file describes the style that is enforced
+by the script, which is based off the LLVM style with modifications closer to
+the default Visual Studio style. See [clang-format style options](
+http://releases.llvm.org/3.6.0/tools/clang/docs/ClangFormatStyleOptions.html)
+for details.
+
+If you see unexpected formatting changes in the code, verify that you are running version 11 or higher of the LLVM tool-chain.
+
+### License Header
+
+The following license header **must** be included at the top of every code file:
+
+```
+// Copyright (c) IO Visor Project
```

Comment:
I agree that there needs to be appropriate copyright notices in each file.

However, Arm hasn't assigned copyright on contributions and to do so would require lawyer time. (Looking at the source code I don't believe any other contributor has either).

Either make it a requirement that there is a copyright line with appropriate ownership (which is all the script checks currently anyway as far as I can tell).